.github/workflows/python-packaging.yml

```diff
@@ -100,3 +117,63 @@ jobs:
     - name: Smoke test ${{ matrix.os }}
       run: python -c 'import tiledbsoma; print(tiledbsoma.libtiledbsoma.__file__)'
       # TODO: more-thorough test
+
+  # ----------------------------------------------------------------
+  # Nominally this is on the nightly schedule
+  publish-to-test-pypi:
+    name: Publish package to TestPyPI
+    needs: wheels
+    runs-on: ubuntu-20.04
+
+    steps:
```

Comment:
@johnkerl The main gap I see right now is that these `publish-to-test-pypi` and `publish-to-pypi` steps will each run on a fresh github worker, separate from the workers where we built the sdist and wheels in prior steps (one of which was a macOS worker, QED). Therefore when they start up they don't yet have the sdist tarball and wheels available on their local filesystem.

We can use the download-artifacts action to fetch them -- this is one of the main use cases of uploading to artifacts (in addition to making them available to download from github yourself). The smoke-test step above this one does so for example, and [another example from the cibuildwheel docs](https://cibuildwheel.readthedocs.io/en/stable/deliver-to-pypi/#github-actions).

Two tricky bits that will follow,

1. download-artifacts extracts the zip of artifacts automatically, and I can never remember what their filesystem layout is going to be on the github worker (they might just end up flat in the working directory). I end up temporarily sticking a `find . -type f` in the script to run and find out from the logs. The files may then need to be reorganized (moved into subdirectories) according to what you need to do with them.
2. The first step in this workflow builds an sdist tarball which is then used as the 'gold master' from which all the wheels are then built. It'd be nice to make sure exactly this sdist tarball is also what we upload to PyPI. In contrast I think the gh-action-pypi-publish is probably building a new sdist tarball from the local source checkout. They should be equivalent, but you never know. Maybe there's some way to fiddle with the config to make sure it publishes the sdist tarball that download-artifacts will localize for us.

Reply:
@mlin thanks for sharing
https://github.com/single-cell-data/TileDB-SOMA/wiki/PyPI-packaging-WIP
which my personal failure to find yesterday cost me an hour of confusion and despair. I'll be sure to consult that link in the future, and I'll work to make it easier for others to find as well.

Now that I'm properly contextualized, I can see why my approach here won't work.

Thanks for your review!

Reply:
Also: @mlin I believe at the very least (as an initial PR) we should remove this entirely:
https://github.com/single-cell-data/TileDB-SOMA/blob/main/.github/workflows/python-ci.yml#L89-L104
as it's a duplicate of code here:
https://github.com/single-cell-data/TileDB-SOMA/blob/main/.github/workflows/python-packaging.yml#L33-L70

I'll put up a separate PR to remove this redundancy